be/src/olap/compaction.cpp

```diff
@@ -181,6 +181,30 @@ OLAPStatus Compaction::gc_unused_rowsets() {
     return OLAP_SUCCESS;
 }
 
+// find the longest consecutive version path in "rowset", from begining.
+OLAPStatus Compaction::find_longest_consecutive_version(vector<RowsetSharedPtr>* rowsets) {
+    if (rowsets->empty()) {
+        return OLAP_SUCCESS;
+    }
+    RowsetSharedPtr prev_rowset = rowsets->front();
+    size_t i = 1;
+    for (; i < rowsets->size(); ++i) {
+        RowsetSharedPtr rowset = (*rowsets)[i];
+        if (rowset->start_version() != prev_rowset->end_version() + 1) {
+            LOG(WARNING) << "There are missed versions among rowsets. "
```

Comment:
Is it appropriate to print the warning log here?
I think this function only deals with specific logic, and the caller can decide whether to print the log. This allows this function to be used in more scenarios.

Reply:
Done